### DISCUSSION TITLE
Controlling the brightness of your laptop screen with i3

### DISCUSSION THREAD

**Discussion by ghost** [i3/i3#4580] [Q&A]
To make screen brightness control on my laptop work I used tutorial which is in below link: 
[URL](https://sourabhtk37.github.io/blog/post/brightness-and-volume-control-i3/#:~:text=set%20Master%20toggle-,Brightness%20Control,-Use%20xev%20again)
Everything cool until the restart. It looks like the chmod file permissions I gave disappear after reboot. Every reboot I have to add them again. 
Is there any way to work around this?

**Comment by dojero** [comment]
You can simply add an exec command for the chmod in your i3 config: exec --no-startup-id sudo chmod [filename]

You will be asked to enter your password for sudo, unless your user has sudo without password privilege for the command.

**Reply by ghost** [reply]
I tried this method but I am not prompted for my password. 
The command simply does not execute.
Is there no other way? Even one that does not use the i3 config file?

**Reply by dojero** [reply]
I am surprised that the command 'simply does not execute'. Did you reload your i3 after you changed the config file? If so, it should at least try to exec the command or give you an error. 

If you don't want to use i3 for this, you can chmod the file in question and then use chattr to make it immutable. The command is chattr +i [filename]. It must be run as a superuser and only a superuser can make that file mutable again. In other words, you'll set the permissions and then make the file unchangeable, even upon a reboot. If you later want to change those permissions, you'll need to chattr -i the file.

**Reply by Airblader** [reply]
There's no shell for the user, of course it won't ask for a password. Where would it?

Changing file permissions from the i3 config also sounds pretty awful. If anything I'd just make the command work (on that sysfs path only!) with passwordless sudo.

I'd also be quite surprised if `chattr +i` works on this path. sysfs isn't a regular file system.

**Reply by dojero** [reply]
I don't know if chattr will work on the /sys file, but I'm unable to find anything in man or elsewhere that says it wouldn't. I wonder what you are saying would set the /sys file apart...is there an attribute that makes chattr fail?

**Comment by dojero** [comment]
My bad, but you can easily use a terminal to run the commmand (exec [your terminal] -e sudo chmod [filename]

**Reply by Airblader** [reply]
Having a terminal open on every login to change a permission so that brightness controls work sounds absolutely insane to me, but everyone makes their own choices.

**Comment by orestisfl** [comment]
Your link is not accessible to me

Have you tried something like [light](http://haikarainen.github.io/light/)?

**Reply by Airblader** [reply]
The link is broken, but the URL being the link text works.

Edit: I edited the post to fix the link now.

**Reply by ghost** [reply]
Noo.. I will try
